apps/contribution/test/contribution.js

```diff
@@ -0,0 +1,196 @@
+
+const getContract = name => artifacts.require(name)
+const { assertRevert } = require('@aragon/test-helpers/assertThrow');
+
+const ZERO_ADDR = '0x0000000000000000000000000000000000000000';
+
+contract('Contribution app', (accounts) => {
+    let kernelBase, aclBase, daoFactory, dao, acl, contribution, token;
+  
+    const root = accounts[0];
+    const member1 = accounts[1];
+
+    before(async() => {
+        kernelBase = await getContract('Kernel').new(true) // petrify immediately
+        aclBase = await getContract('ACL').new()
+        daoFactory = await getContract('DAOFactory').new(kernelBase.address, aclBase.address, ZERO_ADDR);
+        r = await daoFactory.newDAO(root)
+        dao = getContract('Kernel').at(r.logs.filter(l => l.event == 'DeployDAO')[0].args.dao)
+        acl = getContract('ACL').at(await dao.acl())
+    
+        //create dao mamnager permission for coin owner
+        await acl.createPermission(
+            root,
+            dao.address,
+            await dao.APP_MANAGER_ROLE(),
+            root,
+            { from: root }
+        );
+
+        //apps id
+        let appsId = [];
+        appsId[0] = namehash("kredits-contribution");
+        appsId[1] = namehash("kredits-contributor");
+        appsId[2] = namehash("kredits-proposal");
+        appsId[3] = namehash("kredits-token");
+        
+    
+        //get new app instance from DAO
+        let receipt = await dao.newAppInstance(
+            appsId[0],
+            (await Contribution.new()).address,
+            0x0,
+            false,
+            { from: root }
+        )
+        contribution = Contribution.at(
+            receipt.logs.filter(l => l.event == 'NewAppProxy')[0].args.proxy
+        );
+
+        receipt = await dao.newAppInstance(
+            appsId[3],
+            (await getTokenContract('Token').new()).address,
+            0x0,
+            false,
+            { from: root }
+        )
+        token = Token.at(
+            receipt.logs.filter(l => l.event == 'NewAppProxy')[0].args.proxy
+        );
+    
+        //init app
+        await contribution.initialize(appsId);
+    
+        await acl.createPermission(
+            root,
+            contribution.address,
+            await contribution.ADD_CONTRIBUTION_ROLE(),
+            root,
+            { from: root }
+        )
+
+        await acl.createPermission(
+            root,
+            contribution.address,
+            await contribution.VETO_CONTRIBUTION_ROLE(),
+            root,
+            { from: root }
+        );
+
+        //init token (app)
+        await token.initialize(appsId);
+    
+        //create token mint permission for coin owner
+        await acl.createPermission(
+            root,
+            token.address,
+            await token.MINT_TOKEN_ROLE(),
+            root,
+            { from: root }
+        );
+
+        //acl.grantPermission(contribution, token, await token.MINT_TOKEN_ROLE(), {from: root});            
```

Comment:
I have a problem here granting the mint token permission to the contribution app...

Reply:
what do you mean? what's the error?

Reply:
a revert...